src/Engine/Audio/SoundStreamer.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "../linking.hpp"
+#include <thread>
+#include <queue>
+#include <vector>
+#include <mutex>
+
+namespace Audio {
+
+    class SoundFile;
+
+    /// Streams sound data from file.
+    class SoundStreamer {
+        public:
+            /// Constructor.
+            ENGINE_API SoundStreamer();
+            
+            /// Destructor.
+            ENGINE_API ~SoundStreamer();
+
+            /// Worker class used to wrapp threaded sound streaming.
+            class Worker {
```

Comment:
Will Worker be used outside SoundStreamer? If not, it should be private.

Reply:
IDK!!!!